boards/shared/Src/App/App_SharedHeartbeatMonitor.c

```diff
@@ -79,3 +87,18 @@ void App_SharedHeartbeatMonitor_CheckIn(
 {
     heartbeat_monitor->heartbeats_checked_in |= heartbeat;
 }
+
+#if (BOARD_ID == DIM_ID)
```

Comment:
this indicates to me that this should be in a component specific file i.e. DIM file, maybe with a weak function or something

Reply:
I think using a function pointer here might not be the best way to invoke your callback

Reply:
Might be much easier to read if we had the heartbeat tick function return false when a heartbeat signal is not received. We can implement the "callback" in the state machine

Reply:
Rlly wanna be careful with fxn ptrs for next year and make sure we use it only when its needed. Makes the code rlly hard to follow tbh